File: Session13/C++/KthLargestElement.cpp
https://leetcode.com/problems/kth-largest-element-in-an-array/

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        
        priority_queue<int, vector<int>, greater<int>> pq; // min heap
        
        for(int i:nums) {
            pq.push(i);
            if(pq.size() > k) {
                pq.pop();
            }
        }
        
        return pq.top();
    }
};

File: Session13/Java/KthLargestElement.java
https://leetcode.com/problems/kth-largest-element-in-an-array/

class Solution {
    public int findKthLargest(int[] nums, int k) {
        
        PriorityQueue<Integer> pq = new PriorityQueue<>(); // min heap
        
        for(int i:nums) {
            pq.offer(i);
            if(pq.size()>k) {
                pq.poll();
            }
        }
        
        return pq.peek();
    }
}